internal/client.go

```diff
@@ -228,63 +231,92 @@ func GetOrNewRocketMQClient(option ClientOptions, callbackCh chan interface{}) *
 func (c *rmqClient) Start() {
 	//ctx, cancel := context.WithCancel(context.Background())
 	//c.cancel = cancel
-	c.close = false
 	c.once.Do(func() {
-		// TODO fetchNameServerAddr
 		if !c.option.Credentials.IsEmpty() {
 			c.remoteClient.RegisterInterceptor(remote.ACLInterceptor(c.option.Credentials))
 		}
+		// TODO fetchNameServerAddr
 		go func() {}()
 
 		// schedule update route info
 		go func() {
 			// delay
+			ticker := time.NewTicker(_PullNameServerInterval)
+			defer ticker.Stop()
 			time.Sleep(50 * time.Millisecond)
-			for !c.close {
-				c.UpdateTopicRouteInfo()
-				time.Sleep(_PullNameServerInterval)
+			for {
+				select {
+				case <-ticker.C:
+					c.UpdateTopicRouteInfo()
+				case <-c.done:
+					rlog.Info("Stopping update topic route info", nil)
```

Comment:
maybe more info should be provided. for example, clientId group topic etc....

Reply:
The clientID has been added, others info like topic/group are difficult to available because the rmqClient is a common component。